commit 4d25545cc673
Author: Gerald Squelart <gsquelart@mozilla.com>
Date:   Thu Jan 25 16:21:23 2018 +1100

    Bug 1419508 - Return early from CreateMaskLayer if there is no visible data - r=mattwoodrow a=lizzard
    
    MozReview-Commit-ID: 9jWGXA2eGvv
    
    --HG--
    extra : source : c5d8b1ed27228f4bcf45b2746f78dba3795d0868
---
 layout/painting/FrameLayerBuilder.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git layout/painting/FrameLayerBuilder.cpp layout/painting/FrameLayerBuilder.cpp
index 3c5d04260f7e..48ec5db56251 100644
--- layout/painting/FrameLayerBuilder.cpp
+++ layout/painting/FrameLayerBuilder.cpp
@@ -6358,6 +6358,10 @@ ContainerState::CreateMaskLayer(Layer *aLayer,
   gfx::Rect boundingRect = CalculateBounds(newData.mRoundedClipRects,
                                            newData.mAppUnitsPerDevPixel);
   boundingRect.Scale(mParameters.mXScale, mParameters.mYScale);
+  if (boundingRect.IsEmpty()) {
+    // Return early if we know that there is effectively no visible data.
+    return nullptr;
+  }
 
   uint32_t maxSize = mManager->GetMaxTextureSize();
   NS_ASSERTION(maxSize > 0, "Invalid max texture size");
